### PR TITLE
Update external-provisioner for storage provisioner for Kubernetes 1.18

### PR DESCRIPTION
**Pull request by johscheuer**
This PR updates the https://sigs.k8s.io/sig-storage-lib-external-provisioner to version [v5.0.0](https://github.com/kubernetes-sigs/sig-storage-lib-external-provisioner/releases/tag/v5.0.0) to support Kubernetes 1.18 properly (1.18 support was introduces in `5.0.0`). The new version of the lib automatically tries to use leader-election, I added an extra `Role` + `RoleBinding` for the RBAC. For the most use-cases the old version will work but if a controller/operator changes the `PVC` object the storage provisioner will fail with the following error message:

```bash
E0630 10:20:08.514160       1 reflector.go:205] k8s.io/minikube/vendor/github.com/r2d4/external-storage/lib/controller/controller.go:411: Failed to list *v1.PersistentVolumeClaim: v1.PersistentVolumeClaimList: Items: []v1.PersistentVolumeClaim: v1.PersistentVolumeClaim: ObjectMeta: v1.ObjectMeta: readObjectFieldAsBytes: expect :kubectl get storageclassng 912 ...:{},"k:{\"... at {"kind":"PersistentVolumeClaimList","apiVersion":"v1","metadata":{"selfLink":"/api/v1/persistentvolumeclaims","resourceVersion":"1509"},"items":[{"metadata":{"name":"controller-sqlite","namespace":"default","selfLink":"/api/v1/namespaces/default/persistentvolumeclaims/controller-sqlite","uid":"95f218fc-6cee-4fc0-99e7-d0508f698ede","resourceVersion":"1509","creationTimestamp":"2020-06-30T10:15:54Z","annotations":{"volume.beta.kubernetes.io/storage-provisioner":"k8s.io/minikube-hostpath"},"ownerReferences":[{"apiVersion":"iofog.org/v2","kind":"ControlPlane","name":"iofog","uid":"13202d97-bd76-4a01-ba58-c9a0cf43ff21","controller":true,"blockOwnerDeletion":true}],"finalizers":["kubernetes.io/pvc-protection"],"managedFields":[{"manager":"iofog-operator","operation":"Update","apiVersion":"v1","time":"2020-06-30T10:15:54Z","fieldsType":"FieldsV1","fieldsV1":{"f:metadata":{"f:ownerReferences":{".":{},"k:{\"uid\":\"13202d97-bd76-4a01-ba58-c9a0cf43ff21\"}":{".":{},"f:apiVersion":{},"f:blockOwnerDeletion":{},"f:controller":{},"f:kind":{},"f:name":{},"f:uid":{}}}},"f:spec":{"f:accessModes":{},"f:resources":{"f:requests":{".":{},"f:storage":{}}},"f:volumeMode":{}},"f:status":{"f:phase":{}}}},{"manager":"kube-controller-manager","operation":"Update","apiVersion":"v1","time":"2020-06-30T10:15:54Z","fieldsType":"FieldsV1","fieldsV1":{"f:metadata":{"f:annotations":{".":{},"f:volume.beta.kubernetes.io/storage-provisioner":{}}}}}]},"spec":{"accessModes":["ReadWriteOnce"],"resources":{"requests":{"storage":"1Gi"}},"storageClassName":"standard","volumeMode":"Filesystem"},"status":{"phase":"Pending"}}]}
```

With the new  version of the lib everything works as expected.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -85,7 +83,8 @@ require (
 	k8s.io/kubectl v0.0.0
 	k8s.io/kubernetes v1.17.3
 	k8s.io/utils v0.0.0-20200229041039-0a110f9eb7ab // indirect
-	sigs.k8s.io/sig-storage-lib-external-provisioner v4.0.0+incompatible
+	sigs.k8s.io/sig-storage-lib-external-provisioner v4.0.0+incompatible // indirect
```
Comment (johscheuer) [inline go.mod:86]: I tried to run `go mod tidy` (and to remove it manually) but somehow I was not able to remove this import:

```bash
$ go mod tidy
go: finding module for package sigs.k8s.io/sig-storage-lib-external-provisioner/controller/metrics
go: found sigs.k8s.io/sig-storage-lib-external-provisioner/controller/metrics in sigs.k8s.io/sig-storage-lib-external-provisioner v4.1.0+incompatible
```

and

```bash
$ go mod why sigs.k8s.io/sig-storage-lib-external-provisioner/controller/metrics
# sigs.k8s.io/sig-storage-lib-external-provisioner/controller/metrics
k8s.io/minikube/pkg/storage
sigs.k8s.io/sig-storage-lib-external-provisioner/v5/controller
sigs.k8s.io/sig-storage-lib-external-provisioner/v5/controller.test
sigs.k8s.io/sig-storage-lib-external-provisioner/controller/metrics
```

**Comment by johscheuer** [comment]
I just tested it:

```bash
 $ kubectl -n kube-system get po storage-provisioner -o yaml | ag 'image:'
            f:image: {}
    image: johscheuer/storage-provisioner:latest
    image: johscheuer/storage-provisioner:latest
```

and the controller is able to create the PVC:

```bash
I0701 06:38:43.387502       1 leaderelection.go:252] successfully acquired lease kube-system/k8s.io-minikube-hostpath
I0701 06:38:43.390623       1 controller.go:799] Starting provisioner controller k8s.io/minikube-hostpath_minikube_3b784daa-88ed-450c-8cc2-0fe442878e5e!
I0701 06:38:43.391187       1 event.go:281] Event(v1.ObjectReference{Kind:"Endpoints", Namespace:"kube-system", Name:"k8s.io-minikube-hostpath", UID:"3da7db69-b8a1-41ba-a7b3-d01ef82b0d3b", APIVersion:"v1", ResourceVersion:"1950", FieldPath:""}): type: 'Normal' reason: 'LeaderElection' minikube_3b784daa-88ed-450c-8cc2-0fe442878e5e became leader
I0701 06:38:43.493239       1 controller.go:848] Started provisioner controller k8s.io/minikube-hostpath_minikube_3b784daa-88ed-450c-8cc2-0fe442878e5e!
I0701 06:38:43.494436       1 controller.go:1284] provision "default/controller-sqlite" class "standard": started
I0701 06:38:43.504127       1 event.go:281] Event(v1.ObjectReference{Kind:"PersistentVolumeClaim", Namespace:"default", Name:"controller-sqlite", UID:"b04a5cb5-d6a8-4cc3-908d-81d83b79d046", APIVersion:"v1", ResourceVersion:"1177", FieldPath:""}): type: 'Normal' reason: 'Provisioning' External provisioner is provisioning volume for claim "default/controller-sqlite"
I0701 06:38:43.505755       1 controller.go:1392] provision "default/controller-sqlite" class "standard": volume "pvc-b04a5cb5-d6a8-4cc3-908d-81d83b79d046" provisioned
I0701 06:38:43.505803       1 controller.go:1409] provision "default/controller-sqlite" class "standard": succeeded
I0701 06:38:43.506080       1 volume_store.go:212] Trying to save persistentvolume "pvc-b04a5cb5-d6a8-4cc3-908d-81d83b79d046"
I0701 06:38:43.518494       1 volume_store.go:219] persistentvolume "pvc-b04a5cb5-d6a8-4cc3-908d-81d83b79d046" saved
I0701 06:38:43.519411       1 event.go:281] Event(v1.ObjectReference{Kind:"PersistentVolumeClaim", Namespace:"default", Name:"controller-sqlite", UID:"b04a5cb5-d6a8-4cc3-908d-81d83b79d046", APIVersion:"v1", ResourceVersion:"1177", FieldPath:""}): type: 'Normal' reason: 'ProvisioningSucceeded' Successfully provisioned volume pvc-b04a5cb5-d6a8-4cc3-908d-81d83b79d046
```

and 

``bash
$ kubectl get pvc
NAME                STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS   AGE
controller-sqlite   Bound    pvc-b04a5cb5-d6a8-4cc3-908d-81d83b79d046   1Gi        RWO            standard       6m6s
```

**Comment by johscheuer** [comment]
Hm, the failures seem unrelated to me. Is there a way to retrigger them?

**Comment by priyawadhwa** [comment]
Hey @johscheuer I think the test failures are unrelated -- this LGTM

**Pull request merged**